nucypher/characters.py

```diff
@@ -423,20 +425,29 @@ def learn_from_teacher_node(self, eager=True):
             self.log.warning("Can't learn right now: {}".format(e.args[0]))
             return
 
-        rest_url = current_teacher.rest_url()
+        rest_url = current_teacher._crypto_power.power_ups(TLSHostingPower).rest_server.rest_url()
```

Comment:
Yeeeeah.  That's no good.  We need a public method for this.  Let's slap `rest_url` on... `Ursula`?  `VerifiableNode`?